portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -0,0 +1,73 @@
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import com.google.sps.data.Fields;
+import com.google.sps.data.Video;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+// Servlet for testing queries.
+@WebServlet("/data")
+public class DataServlet extends HttpServlet {
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+        String id = request.getParameter(Fields.VIDEO_ID);
+        if(id == null){
+            response.getWriter().println("AAAAAAAAAAAAAAAAAAAAAA");
+        }
+        Key idAsKey = KeyFactory.createKey("Video", id);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Query query = new Query("Video")
+        .setFilter(new FilterPredicate("videoId", FilterOperator.EQUAL, id));
+        PreparedQuery preparedQuery = datastore.prepare(query);
+        Entity video = preparedQuery.asSingleEntity();
+        String videoObjectJson = video.getProperty("videoObjectAsJson").toString();
```

Comment:
It might be a good idea to add "videoObjectAsJson" to the PropertyNames class and then do video.getProperty(PropertyNames.videoObjectAsJson). The point of this is that PropertyNames.videoObjectAsJson will error if you mispell the property name. getProperty will silently return null if property strings are mispelled (which is dangerous).

Reply:
Have you checked this line to make sure it works? I'm not sure if toString() will convert the object properly (it may just return the object address). Instead, you may want to just do (String) video.getProperty(Property.videoObjectAsJson);